src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -495,4 +500,135 @@ public void close() throws Exception {
         });
     }
 
+    // Support methods for REST API's
+
+    public List<String> getInspectionHooksIds() throws Exception {
+
+        return this.txControl.supports(() -> {
+
+            List<InspectionHookEntity> elements = this.utils.txInspectionHookEntities();
+            List<String> elementList = new ArrayList<String>();
+
+            if (!elements.isEmpty()) {
+                elementList = elements.stream().map(InspectionHookEntity::getHookId).collect(Collectors.toList());
+            }
+
+            return elementList;
+        });
+    }
+
+    public Element updateInspectionPort(InspectionPortElement inspectionPort) throws Exception {
+
+        InspectionPortEntity inspectionPortEntity = (InspectionPortEntity) getInspectionPort(inspectionPort);
+        if (inspectionPortEntity == null) {
+            String msg = String.format("inspection port element does not exists id: %s\n",
+                    inspectionPort.getElementId());
+            LOG.error(msg);
+            throw new IllegalArgumentException(msg);
+        }
+
+        return this.txControl.required(() -> {
+            inspectionPortEntity.setEgressPort((NetworkElementEntity) inspectionPort.getEgressPort());
```

Comment:
where do we validate the egress and ingress ports exist?

Since we have unit tests for nsc plugin...please use the tests to validate all these scenarios